unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/WorldChatWindowHUD/PromoteChannelsToastComponentController.cs

```diff
@@ -0,0 +1,72 @@
+using DCL.Helpers;
+using System;
+
+namespace DCL.Chat.HUD
+{
+    public class PromoteChannelsToastComponentController : IDisposable
+    {
+        internal const string PLAYER_PREFS_PROMOTE_CHANNELS_TOAS_DISMISSED_KEY = "PromoteChannelsToastDismissed";
+
+        private readonly IPromoteChannelsToastComponentView view;
+        private readonly DataStore dataStore;
+
+        private BaseVariable<bool> isPromoteToastVisible => dataStore.channels.isPromoteToastVisible;
+
+        public PromoteChannelsToastComponentController(
+            IPromoteChannelsToastComponentView view,
+            DataStore dataStore)
+        {
+            this.view = view;
+            this.view.OnClose += Dismiss;
+            this.view.Hide();
+
+            this.dataStore = dataStore;
+
+            CommonScriptableObjects.rendererState.OnChange += RendererState_OnChange;
+        }
+
+        private void RendererState_OnChange(bool current, bool previous)
+        {
+            if (!current)
+                return;
+
+            CommonScriptableObjects.rendererState.OnChange -= RendererState_OnChange;
+            isPromoteToastVisible.OnChange += OnToastVisbile;
+
+            if (!PlayerPrefsUtils.GetBool(PLAYER_PREFS_PROMOTE_CHANNELS_TOAS_DISMISSED_KEY, false))
+                isPromoteToastVisible.Set(true, notifyEvent: true);
+            else
+                isPromoteToastVisible.Set(false, notifyEvent: true);
+        }
+
+        private void Dismiss()
+        {
+            isPromoteToastVisible.Set(false);
+            PlayerPrefsUtils.SetBool(PLAYER_PREFS_PROMOTE_CHANNELS_TOAS_DISMISSED_KEY, true);
+            PlayerPrefsUtils.Save();
+        }
+
+        private void OnToastVisbile(bool isVisible, bool _)
+        {
+            if (isVisible)
+                view.Show();
+            else
+            {
+                isPromoteToastVisible.OnChange -= OnToastVisbile;
+                view.Hide();
+            }
+        }
+
+        public void Dispose()
+        {
+            if (view != null)
+            {
+                view.OnClose -= Dismiss;
+                view.Dispose();
+            }
+
+            CommonScriptableObjects.rendererState.OnChange -= RendererState_OnChange;
```

Comment:
you can inject `CommonScriptableObjects.rendererState` instead of the static usage